src/util/antoraSupport.ts

```diff
@@ -0,0 +1,36 @@
+import vscode from 'vscode'
+import fs from 'fs'
+import yaml from 'js-yaml'
+import path from 'path'
+
+export default class AntoraSupport {
+  public async isDetected (): Promise<boolean> {
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 10)
+    return antoraYmlFiles.length > 0
+  }
+
+  public async findModulesDirPath (textDocument: vscode.TextDocument): Promise<any> {
+    const pathToAsciidocFile = textDocument.uri.fsPath
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 100)
+    // check if antora.yml file matches with textDocument content source root
+    try {
+      for (let i = 0; i < antoraYmlFiles.length; i++) {
+        if (pathToAsciidocFile.startsWith(path.dirname(antoraYmlFiles[i].path))) {
```

Comment:
(nitpick) you can use `for ... of` (simpler syntax):


```suggestion
      for (const antoraYmlFile of antoraYmlFiles) {
        if (pathToAsciidocFile.startsWith(path.dirname(antoraYmlFile.path))) {
```